src/main/java/li/cil/oc/api/network/SimpleComponent.java

```diff
@@ -1,114 +0,0 @@
- * Classes implementing this interface will be expanded with the methods
- * required for them to function as native block components (say, like the
- * screen or keyboard). This means functions in the <tt>Environment</tt>
- * interface have to created using a class transformer. If any of the methods
- * already exist, this will fail! If things don't work, check your logs, first.
- * <p/>
- * To expose methods to OC, tag them with {@link li.cil.oc.api.machine.Callback}
- * and have them use the according signature (see the documentation on the
- * <tt>Callback</tt> annotation).
- * <p/>
- * Alternatively, implement {@link li.cil.oc.api.network.ManagedPeripheral} in
- * addition to this interface, to make methods available ComputerCraft style.
- * <p/>
- * So, in short:
- * <ul>
- * <li>Implement this interface on a tile entity that should expose
- * methods to computers.</li>
- * <li>Annotate methods with <tt>Callback</tt> so they exported.</li>
- * <li>Alternatively/additionally implement <tt>ManagedPeripheral</tt> to
- * provide methods via a list of names and single callback method.</li>
- * </ul>
- * <p/>
- * For example:
- * <pre>
- *     {@literal @}Optional.Interface(iface = "li.cil.oc.api.network.SimpleComponent", modid = "OpenComputers")
- *     public class TileEntityMyFancyThing extends TileEntity
- *            implements SimpleComponent
- *     {
- *         {@literal @}Override
- *         public String getComponentName() {
- *             return "fancy_thing";
- *         }
- *
- *         {@literal @}Callback
- *         {@literal @}Optional.Method(modid = "OpenComputers")
- *         public Object[] greet(Context context, Arguments args) {
- *             return new Object[]{String.format("Hello, %s!", args.checkString(0))};
- *         }
- *     }
- * </pre>
- * Using the alternative method to provide methods:
- * <pre>
- *     {@literal @}Optional.InterfaceList({
- *         {@literal @}Optional.Interface(iface = "li.cil.oc.api.network.SimpleComponent", modid = "OpenComputers"),
- *         {@literal @}Optional.Interface(iface = "li.cil.oc.api.network.ManagedPeripheral", modid = "OpenComputers")
- *     })
- *     public class TileEntityMyFancyThing extends TileEntity
- *            implements SimpleComponent, ManagedPeripheral
- *     {
- *         {@literal @}Override
- *         public String getComponentName() {
- *             return "fancy_thing";
- *         }
- *
- *         public String[] methods() {
- *             return new String[] {"greet"};
- *         }
- *
- *         {@literal @}Optional.Method(modid = "OpenComputers")
- *         public Object[] invoke(String method, Context context, Arguments args) {
- *             if ("greet".equals(method)) {
- *                 return new Object[]{String.format("Hello, %s!", args.checkString(0))};
- *             } else {
- *                 throw new NoSuchMethodException();
- *             }
- *         }
- *     }
- * </pre>
- */
-public interface SimpleComponent {
```

Comment:
I'm open to reintroduce the SimpleComponent in the future, indeed the removal of the stubs was probably rushed as, even without class transformer support (which I have no experience with), modders could call them manually.

Reply:
As far as I know, JavaScript-based coremods don't allow you to target arbitrary classes, only a pre-determined list. If that is the case, a new approach would be necessary. We have Capabilities now (if Forge didn't remove them), so that might be the right way to go...

Reply:
Ah, I see, yeah, that makes it more complicated/a bigger change than I expected. Switching this over to (potentially event injected) capabilities sounds like a good new approach in a next step.